python/cudf/cudf/core/udf/utils.py

```diff
@@ -213,6 +230,31 @@ def _get_kernel(kernel_string, globals_, sig, func):
     globals_["f_"] = f_
     exec(kernel_string, globals_)
     _kernel = globals_["_kernel"]
-    kernel = cuda.jit(sig)(_kernel)
+    kernel = cuda.jit(sig, link=files, extensions=arg_handlers)(_kernel)
 
     return kernel
+
+
+launch_arg_getters: Dict[Any, Any] = {}
+
+
+def _launch_arg_from_col(col):
+    getter = launch_arg_getters.get(col.dtype)
+    if getter:
+        data = getter(col)
+    else:
+        data = col.data
+
+    mask = col.mask
+    if mask is None:
+        return data
+    else:
+        return data, mask
```

Comment:
A pattern I see reoccurring in a few places is the use of different return types based on masks / attributes. That feels a little awkward. For most functions, I prefer to see a single return type regardless of the input types. Perhaps use `None` for nonexistent masks.